### PR TITLE
Dry run benchmarks using tox to discover potential bugs

### PR DESCRIPTION
**Pull request by JonathanWenger**
This PR adds a `benchmarks` environment to `tox` which makes a quick dry run of the benchmarks in the current environment defined by tox. This checks whether benchmarks execute properly or not (see also https://asv.readthedocs.io/en/stable/writing_benchmarks.html#running-benchmarks-during-development).

### CODE REVIEW

**Comment by JonathanWenger** [comment]
This closes #112.

**Comment by JonathanWenger** [comment]
It's probably a good idea early on where interfaces might be changed or people add new benchmarks. This is somewhat different from the build, docs and tests which are effected by nearly all code changes. Long-term this will likely only extend tox runtime with little benefit.

**Comment by nathanaelbosch** [comment]
True, I also think we should add the `benchmarks` env to the CI workflow!

File: tox.ini [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
-envlist = py3,docs
+envlist = py3, docs, benchmarks
 
 [travis]
 python =
-  3.6: py36, docs
-  3.7: py37
-  3.8: py38
+  3.6: py36, docs, benchmarks
+  3.7: py37, docs, benchmarks
```
Comment (nathanaelbosch) [inline tox.ini:12]: To not mix too many things regarding the ongoing discussion in #127, I would propose to stick to the previous behaviour to merge this PR, so adding `benchmarks` just to the `envlist` and 3.6, but not to 3.7 and 3.8. Depending on the conlcusion in #127 this could then be fixed alter in a different PR.

Comment (JonathanWenger) [reply tox.ini]: Good point. Reverted back to the previous behavior.

**Comment by nathanaelbosch** [comment]
Sorry if my previous statement was confusing. In order to actually run the benchmarks in Travis, could you add `benchmarks` to `3.6` in line 11 again?
